### PR TITLE
chore:system user and events created by userid migrations

### PR DESCRIPTION
**Pull request by daveleek**
## About the changes

Migrations for:
- Adds column created_by_user_id to events
- Adds column is_system to users
- Inserts unleash_system_user id -1337 to users 

includes `is_system: false` in the activeUsers where filter

Tested by running:
`
select * into users_pre_check from users where id > -1;
delete from users where id > -1;
`
before starting unleash, then inspecting users table after unleash has started and verifying that an 'admin' user has been created.

### CODE REVIEW

**Review by chriswk — APPROVED** [review]
Nice. :)

**Review by chriswk — CHANGES_REQUESTED** [review]
Once you've confirmed that this doesn't break creation of initial admin user I'm all for this.

File: src/migrations/20231212094044-event-created-by-user-id.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+exports.up = function (db, callback) {
+    db.runSql(
+        `
+        ALTER TABLE events ADD COLUMN IF NOT EXISTS created_by_user_id INTEGER;
```
Comment (chriswk) [inline src/migrations/20231212094044-event-created-by-user-id.js:6]:
```suggestion
        ALTER TABLE events ADD COLUMN IF NOT EXISTS created_by_user_id INTEGER;
        CREATE INDEX events_created_by_user_id_idx ON events(created_by_user_id);
```

File: src/migrations/20231212094044-event-created-by-user-id.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        `,
+        callback,
+    );
+};
+
+exports.down = function (db, callback) {
+    db.runSql(
+        `
+        ALTER TABLE events DROP COLUMN IF EXISTS created_by_user_id;
```
Comment (chriswk) [inline src/migrations/20231212094044-event-created-by-user-id.js:15]:
```suggestion
        DROP INDEX IF EXISTS events_created_by_user_id_idx;
        ALTER TABLE events DROP COLUMN IF EXISTS created_by_user_id;
```

**Review by chriswk — APPROVED** [review]
Nice. I assume the `where(is_system: false)` will solve the creating admin user.

**Review by nunogois — COMMENTED** [review]
Adding a test like:
 - `favor-permission-name-over-id.e2e.test.ts`
 - `dedupe-permissions.e2e.test.ts`

Could potentially be useful just to validate the migration behavior.

File: src/migrations/20231212094343-unleash-system-user.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+exports.up = function (db, callback) {
+    db.runSql(
+        `
+        ALTER TABLE users ADD COLUMN IF NOT EXISTS is_system BOOLEAN NOT NULL DEFAULT FALSE;
+        INSERT INTO users
+          (id, name, username, email, created_by, is_system) 
+        VALUES
+          (-1337, 'Used by unleash internally for performing system actions that have no user', 'unleash_system_user', 'system@getunleash.io', -1337, true);
```
Comment (nunogois) [inline src/migrations/20231212094343-unleash-system-user.js:10]: I'm a bit unsure about using such a long name in the UI. Does it break anything visually? We might just want a simpler name like "Unleash System" or something.

```suggestion
          (-1337, 'This is a system user utilized internally by Unleash to carry out automated system actions that do not require direct user intervention', 'unleash_system_user', 'system@getunleash.io', -1337, true);
```

Comment (daveleek) [reply src/migrations/20231212094343-unleash-system-user.js:10]: Thanks @nunogois that's a very good point!

**Comment by daveleek** [comment]
Test added, I've updated the username of the system user @nunogois

File: src/migrations/20231222071533-unleash-system-user.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+exports.up = function (db, callback) {
+    db.runSql(
+        `
+        ALTER TABLE users ADD COLUMN IF NOT EXISTS is_system BOOLEAN NOT NULL DEFAULT FALSE;
+        INSERT INTO users
+          (id, name, username, email, created_by_user_id, is_system) 
+        VALUES
+          (-1337, 'Unleash System', 'unleash_system_user', 'system@getunleash.io', -1337, true);
```
Comment (nunogois) [inline src/migrations/20231222071533-unleash-system-user.js:10]: Even though I like `-1337`, I'm wondering if we should just use `-1` instead.

File: src/test/e2e/system-user-migration.test.ts [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    await dbm.reset();
+
+    expect(userResults.length).toEqual(2);
+    expect(userResults[0].is_system).toEqual(true);
+    expect(userResults[0].id).toEqual(-1337);
+    expect(userResults[0].username).toEqual('unleash_system_user');
+    expect(userResults[1].is_system).toEqual(false);
+    expect(userResults[1].id).toEqual(1);
+    expect(userResults[1].username).toEqual('testperson');
+});
```
Comment (nunogois) [inline src/test/e2e/system-user-migration.test.ts:76]: Nice 🏅

**Review by nunogois — APPROVED** [review]
LGTM!

**Pull request merged**